pkg/products/marin3r/reconciler_test.go

```diff
@@ -233,3 +237,253 @@ func strPtr(str string) *string {
 func getLogger() l.Logger {
 	return l.NewLoggerWithContext(l.Fields{l.ProductLogContext: integreatlyv1alpha1.ProductMarin3r})
 }
+
+func TestReconcileEnvoyConfigRevisionsDeletion(t *testing.T) {
+	scheme, err := getBuildScheme()
+	_ = v1alpha1.SchemeBuilder.AddToScheme(scheme)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		name         string
+		serverClient func() k8sclient.Client
+		installation *integreatlyv1alpha1.RHMI
+		want         integreatlyv1alpha1.StatusPhase
+		testFunction func(context.Context, k8sclient.Client) error
+	}{
+		{
+			name: "Basic V2 config revision gets deleted",
+			serverClient: func() k8sclient.Client {
+				return fakeclient.NewFakeClientWithScheme(scheme, &v1alpha1.EnvoyConfigRevisionList{
+					Items: []v1alpha1.EnvoyConfigRevision{
+						{
+							ObjectMeta: v1.ObjectMeta{
+								Finalizers: []string{
+									"envoyconfigrevisions.marin3r.3scale.net",
+								},
+								Labels: map[string]string{
+									"marin3r.3scale.net/envoy-api": "v2",
+								},
+								Namespace: "redhat-rhoam-3scale",
+								Name:      "apicast-ratelimit",
+							},
+							TypeMeta: v1.TypeMeta{
+								APIVersion: "V2",
+							},
+							Spec:   v1alpha1.EnvoyConfigRevisionSpec{},
+							Status: v1alpha1.EnvoyConfigRevisionStatus{},
+						},
+					},
+				})
+			},
+			installation: &integreatlyv1alpha1.RHMI{
+				Spec: integreatlyv1alpha1.RHMISpec{
+					NamespacePrefix: "redhat-rhoam-",
+				},
+			},
+			want:         integreatlyv1alpha1.PhaseCompleted,
+			testFunction: confirmThatConfigRevisionsAreCleared,
+		},
+		{
+			name: "Basic V2 config revision gets finalizers removed while having deletion timestamp",
+			serverClient: func() k8sclient.Client {
+				return fakeclient.NewFakeClientWithScheme(scheme, &v1alpha1.EnvoyConfigRevisionList{
+					Items: []v1alpha1.EnvoyConfigRevision{
+						{
+							ObjectMeta: v1.ObjectMeta{
+								Finalizers: []string{
+									"envoyconfigrevisions.marin3r.3scale.net",
+								},
+								Labels: map[string]string{
+									"marin3r.3scale.net/envoy-api": "v2",
+								},
+								Namespace:         "redhat-rhoam-3scale",
+								Name:              "apicast-ratelimit",
+								DeletionTimestamp: &v1.Time{time.Now()},
```

Comment:
looks like this line is what `make code/fix` is complaining about

Reply:
thanks, I've refactored, let's see how it looks now

Reply:
looks like format passed